src/sentry/scim/endpoints/users.py

```diff
@@ -0,0 +1,211 @@
+from sentry.models import AuditLogEntryEvent, AuthIdentity, OrganizationMember
+from sentry.signals import member_invited
+
+from .utils import SCIM_API_ERROR, SCIMEndpoint, parse_filter_conditions
+
+ERR_ONLY_OWNER = "You cannot remove the only remaining owner of the organization."
+
+SCIM_404_USER_RES = {
+    "schemas": [SCIM_API_ERROR],
+    "detail": "User not found.",
+}
+
+SCIM_409_USER_EXISTS = {
+    "schemas": ["urn:ietf:params:scim:api:messages:2.0:Error"],
+    "detail": "User already exists in the database.",
+}
+
+
+class SCIMUserSerializer(serializers.Serializer):
+    email = AllowedEmailField(max_length=75, required=True)
+    role = serializers.ChoiceField(choices=roles.get_choices(), required=True)
+    sendInvite = serializers.BooleanField(required=False, default=True, write_only=True)
+
+    def validate_email(self, email):
+        queryset = OrganizationMember.objects.filter(
+            Q(email=email) | Q(user__email__iexact=email, user__is_active=True),
+            organization=self.context["organization"],
+        )
+
+        if queryset.filter.exists():
+            raise serializers.ValidationError("User already exists in the database.")
+        return email
+
+
+class OrganizationSCIMUserDetails(SCIMEndpoint, OrganizationMemberDetailsEndpoint):
+    def _get_member(self, organization, member_id):
+        try:
+            member = OrganizationMember.objects.get(organization=organization, id=member_id)
+        except OrganizationMember.DoesNotExist:
+            raise ResourceDoesNotExist
+        except AssertionError as error:
+            if str(error) == "value too large":
+                raise ResourceDoesNotExist
+            raise error
+        return member
+
+    def get(self, request, organization, member_id):
+        try:
+            member = self._get_member(organization, member_id)
+        except ResourceDoesNotExist:
+            return Response(SCIM_404_USER_RES, status=404)
+
+        context = serialize(member, serializer=OrganizationMemberSCIMSerializer())
+        return Response(context)
+
+    def patch(self, request, organization, member_id):
+        try:
+            member = self._get_member(organization, member_id)
+        except ResourceDoesNotExist:
+            return Response(SCIM_404_USER_RES, status=404)
+
+        for operation in request.data.get("Operations", []):
+            # we only support setting active to False which deletes the orgmember
+            if operation["value"]["active"] is False:
+                audit_data = member.get_audit_log_data()
+                if self._is_only_owner(member):
+                    return Response({"detail": ERR_ONLY_OWNER}, status=403)
+                with transaction.atomic():
+                    AuthIdentity.objects.filter(
+                        user=member.user, auth_provider__organization=organization
+                    ).delete()
+                    member.delete()
+                self.create_audit_entry(
```

Comment:
Should the audit entry also be part of the transaction?

Reply:
where i copied this code from, it isn't -- https://github.com/getsentry/sentry/blob/cffe2ad6f3b07202d3b7da43493046610586df53/src/sentry/api/endpoints/organization_member_details.py#L282

it seems like in most places we keep the audit entry out of the proceeding transaction? unsure what the reasoning for that is.